app/src/org/commcare/activities/components/FormNavigationUI.java

```diff
@@ -77,30 +85,78 @@ public static void updateNavigationCues(CommCareActivity activity,
 
     private static void setDoneState(ImageButton nextButton,
                                      Context context,
-                                     ProgressBar progressBar,
-                                     int totalQuestions) {
-        nextButton.setImageResource(R.drawable.icon_chevron_right_attnpos);
-        nextButton.setTag(FormEntryActivity.NAV_STATE_DONE);
+                                     final ClippingFrame finishButton,
+                                     FormNavigationController.NavigationDetails details,
+                                     ProgressBar progressBar) {
+        if(nextButton.getTag() == null) {
+            setFinishVisible(finishButton);
+        }else if(!FormEntryActivity.NAV_STATE_DONE.equals(nextButton.getTag())) {
+            nextButton.setTag(FormEntryActivity.NAV_STATE_DONE);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
+                inflateFinishButton(context, finishButton);
+            } else {
+                setFinishVisible(finishButton);
+            }
+        }
 
         progressBar.setProgressDrawable(context.getResources().getDrawable(R.drawable.progressbar_full));
-        progressBar.setProgress(totalQuestions);
+        progressBar.setProgress(details.totalQuestions);
 
         Log.i("Questions", "Form complete");
     }
 
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
+    private static void inflateFinishButton(Context context,
```

Comment:
"inflate" was a confusing word for me, because the view isn't being inflated in the android sense, but rather the view showing an inflation animation. Maybe a name like "animateFinishButtonInflation"?